src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java

```
@@ -1,16 +1,28 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.MovieDao;
+import mate.academy.hibernate.relations.dao.impl.MovieDaoImpl;
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
+    private final MovieDao movieDao;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
+        this.movieDao = new MovieDaoImpl(sessionFactory);
+    }
+
     @Override
     public Movie add(Movie movie) {
-        return null;
+
+        return movieDao.add(movie);
     }
 
     @Override
     public Movie get(Long id) {
-        return null;
+
+        return movieDao.get(id).orElse(null);
```

Consider handling the case where the `Optional` is empty more explicitly, possibly by throwing an exception or returning a default value, instead of returning `null`. This can help avoid potential `NullPointerExceptions` in the calling code.